packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/OperationInterpreterViewSwitch.java

```diff
@@ -101,6 +102,21 @@ public Optional<VariableManager> caseIf(If ifOperation) {
         }
     }
 
+    @Override
+    public Optional<VariableManager> caseFor(For forOperation) {
+        VariableManager childVariableManager = this.variableManager.createChild();
+        Optional<List<Object>> optionalList = this.interpreter.evaluateExpression(this.variableManager.getVariables(), forOperation.getExpression()).asObjects();
+        if (optionalList.isPresent()) {
+            for (Object object : optionalList.get()) {
+                childVariableManager.put(forOperation.getIteratorName(), object);
```

Comment:
I guess it's ok not to recreate a variable manager on each iteration, just change an iterator value...

Reply:
I'll update this part to instead create a new variable manager on each iteration, child operations may add additional values in the variable manager and it may create "variable leak" where a subtree of the variable manager could be visible in another iteration.